### PR TITLE
use tool updates also on tests on PRs

### PR DESCRIPTION
**Pull request by Bischoff**
## What does this PR change?

Before this PR:

* on a tests on PRs deployment:
```
[root@suma-pr2-min-centos7 yum.repos.d]# ls *_repo.repo 
tools_pool_repo.repo
```
* on Uyuni master deployment:
```
[root@uyuni-master-min-centos7 yum.repos.d]# ls *_repo.repo 
tools_pool_repo.repo tools_update_repo.repo
```

This PR enables `tools_update_repo.repo` also for tests on PRs on CentOS.

### CODE REVIEW

**Review by srbarrios — APPROVED** [review]

**Review by ktsamis — DISMISSED** [review]
I think you are not initialising this with true by default anymore, it might have consequences down the line for these states. Also not sure if the if statement together with the or statement are syntactically and logically correct but I assume you tested this. For now I would say initialise the product_version with the uyuni-master value that it was before and have also Ricardo check this

**Comment by Bischoff** [comment]
> I think you are not initialising this with true by default anymore, it might have consequences down the line for these states. Also not sure if the if statement together with the or statement are syntactically and logically correct but I assume you tested this. For now I would say initialise the product_version with the uyuni-master value that it was before and have also Ricardo check this

default('', true) means the default is '' i think

**Review by rjmateus — COMMENTED** [review]
one comment about the statement logic

File: salt/repos/default.sls [hunk 1]
```diff
@@ -578,7 +578,7 @@ tools_update_repo:
     - priority: 98
     - require:
       - cmd: galaxy_key
-{% elif 'uyuni-master' in grains.get('product_version') | default('', true) %}
+{% elif 'uyuni-master' in grains.get('product_version', '') or 'uyuni-pr' in grains.get('product_version', '') %}
```
[the remark this answers is not in the document]
Comment (Bischoff) [reply salt/repos/default.sls:581]: doesn't `grain.get(x, '')` default to `''` when grain x does not exist?

[the remark this answers is not in the document]
Comment (rjmateus) [reply salt/repos/default.sls:581]: you are correct. please ignore my comment

**Review by juliogonzalez — APPROVED** [review]

File: salt/repos/default.sls (hunk 1 above)
[the remark this answers is not in the document]
Comment (Bischoff) [reply salt/repos/default.sls:581]: No worries. I was not completely sure myself.

**Comment by jordimassaguerpla** [comment]
This is adding a repo from the openSUSE mirror infrastructure, which means adding a new point of failure :(

It would have been better to not add this via sumaform but instead use the publish script https://github.com/uyuni-project/uyuni/blob/master/susemanager-utils/testing/automation/publish-rpms.sh in the build step of jenkins , as we do for other packages.

**Comment by Bischoff** [comment]
> This is adding a repo from the openSUSE mirror infrastructure, which means adding a new point of failure :(
> 
> It would have been better to not add this via sumaform but instead use the publish script https://github.com/uyuni-project/uyuni/blob/master/susemanager-utils/testing/automation/publish-rpms.sh in the build step of jenkins , as we do for other packages.

I just followed what was done all along in sumaform.

If you want to change that, please do it for this whole HCL code (all repos in all branches), not only for this specific case. And as it's a violent change, please coordinate with @rjmateus .

**Pull request merged**